pkg/monitor/cluster/certificateexpirationstatuses.go

```diff
@@ -46,7 +47,12 @@ func (mon *Monitor) emitCertificateExpirationStatuses(ctx context.Context) error
 		})
 	}
 
-	if dns.IsManagedDomain(mon.oc.Properties.ClusterProfile.Domain) {
+	host, err := getHostFromAPIURL(mon.oc.Properties.APIServerProfile.URL)
+	if err != nil {
+		return err
+	}
+
+	if dns.IsManagedDomain(host) {
```

Comment:
Do we need any changes to `dns.IsManagedDomain()` as well, or is the change just moving from checking `mon.oc.Properties.ClusterProfile.Domain` to `mon.oc.Properties.APIServerProfile.URL`?

Reply:
There isn't any changes required to `dns.IsManaged()` as it just checks if the provided string (host of API URL in our case) ends with a managed domain. The change to `mon.oc.Properties.APIServerProfile.URL` was required as `mon.oc.Properties.ClusterProfile.Domain` would not always end with domain at end of it.